### ISSUE TITLE
MSSQL does not return DATEONLY but full Date

### ISSUE THREAD

**Issue by alexvicegrab** [sequelize/sequelize#9325]
## What are you doing?

I have a User model using:

```
 type: DataTypes.DATEONLY,
```

But this is returning full DATE when using MSSQL, but is fine when using SQLITE

## What do you expect to happen?

I expected to see this string instead:
```
1969-06-07
```

## What is actually happening?

However, when I start getting records, these are returned as full date objects (not a DATEONLY string):
e.g.:
```
1969-06-07T00:00:00.000Z
```

__Dialect:__ mssql
__Dialect version:__: N.A.
__Database version:__ : Latest Azure SQL Server
__Sequelize version:__: Sequelize CLI [Node: 9.3.0, CLI: 4.0.0, ORM: 4.37.4]
__Tested with latest release:__ No (If yes, specify that version)

**Comment by seangibbens** [comment]
I am seeing a similar issue when using Postgres.

My model:
```js
module.exports = (sequelize, DataTypes) => {
  const User = sequelize.define('User', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
      allowNull: false,
      unique: true
    },
    startDate: {
      // field in question
      type: DataTypes.DATEONLY,
      field: 'start_date',
      allowNull: false
    }
  }, {
    tableName: 'users',
    paranoid: true
  })

  return User
}
```

When calling `User.findAll()`, correct `startDate` DATEONLY values are being returned.
e.g. `2018-04-20`

However, when calling `User.findOrCreate()` full date + timestamps are being returned in the `startDate` DATEONLY field.
e.g. `2018-04-20T00:00:00.000Z`

Dialect: postgres
Dialect version: 7.4.1
Database version: 9.6
Sequelize version: 4.37.5

**Comment by mmiszy** [comment]
Was this fixed when https://github.com/sequelize/sequelize/issues/9468 was closed?

**Comment by alexvicegrab** [comment]
Good question, I will need to check.

**Comment by sushantdhiman** [comment]
Fix is available in `v5.0.0-beta.7` https://github.com/sequelize/sequelize/blob/master/docs/upgrade-to-v5.md#500-beta7
